src/publish.ts

```diff
@@ -27,16 +28,24 @@ export async function publish(config: PluginConfig, context: Context) {
         s3Bucket = config.s3Bucket[context.branch.name]
     }
 
+    const s3BucketWithResolvedVariables = template(s3Bucket)(
+        {
+            branch: context.branch.name,
+            lastRelease: context.lastRelease,
+            nextRelease: context.nextRelease,
+        }
+    )
+
     const [
         bucketName,
         ...bucketPrefixes
-    ] = s3Bucket?.split(path.sep) ?? []
+    ] = s3BucketWithResolvedVariables.split(path.sep)
 
     const bucketPrefix = bucketPrefixes.join(path.sep).replace(/\$([_a-z]+\w*)|\$\{(\w*)\}/giu, (match, p1, p2) => {
         return process.env[p1 || p2] ?? match
     })
 
-    if (!bucketName) {
+    if (!bucketName || bucketName === '') {
```

Comment:
This is not necessary as empty string is false value

Reply:
I'm not sure to understand, but for me it's a necessary check (see below)

![image](https://user-images.githubusercontent.com/16322590/223146044-7a5e8ca7-2e6d-4729-a945-d5179f77704d.png)

Reply:
Yes we want to go into that if if bucketName is empty string and !bucketName or `!""` will return true and bucketName === '' is unnecessary

Reply:
Oh yes ! Well seen :) I change